cmd/minishift/cmd/root.go

```diff
@@ -254,3 +282,39 @@ func performPostUpdateExecution(markerPath string) error {
 
 	return nil
 }
+
+func ensureAllInstanceConfigPath(configPath string) {
+	configDir := filepath.Dir(configPath)
+	if err := os.MkdirAll(configDir, 0777); err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating directory: %s", configDir))
+	}
+}
+
+func createDirForMinishiftRun(dirs *minishiftConfig.MinishiftDirs) {
+	dirPaths := reflect.ValueOf(*dirs)
+
+	for i := 0; i < dirPaths.NumField(); i++ {
+		path := dirPaths.Field(i).Interface().(string)
+		if err := os.MkdirAll(path, 0777); err != nil {
+			atexit.ExitWithMessage(1, fmt.Sprintf("Error creating directory: %s", path))
+		}
+	}
+}
+
+//If there is no active profiles we need to set
+// minishift as the default profile if minishift profile exists
+func setDefaultActiveProfile() {
+	if minishiftConfig.AllInstancesConfig == nil {
+		atexit.ExitWithMessage(1, "Error: All instance config is not initialized")
+	}
+	activeProfile := profileActions.GetActiveProfile()
+	if activeProfile == "" {
+		err := profileActions.SetActiveProfile(constants.DefaultProfileName)
```

Comment:
ditto as before. This code this feels unnecessary complex in the way it is written.

```diff
-		err := profileActions.SetActiveProfile(constants.DefaultProfileName)
+		err := profileActions.SetDefaultProfileActive()
```

Reply:
@gbraad I am not sure about this as this is just one line and I do not expect it to grow to more lines.

Reply:
@LalatenduMohanty It is not about a number of lines of code, but about reasoning about code.